File: challenges/1337/206reverseLL.swift
/**
 * Reverse a singly linked list.
 * Definition for singly-linked list.
 * public class ListNode {
 *     public var val: Int
 *     public var next: ListNode?
 *     public init(_ val: Int) {
 *         self.val = val
 *         self.next = nil
 *     }
 * }
 */
class Solution {
    func reverseList(_ head: ListNode?) -> ListNode? {
        guard var curr = head else { return nil }
        var temp: ListNode?
        while curr.next != nil {
            var next = curr.next!
            curr.next = temp
            temp = curr
            curr = next 
        }
        curr.next = temp
        return curr
    }
}

File: challenges/1337/206reverseLL.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseList = function(head) {
    [prev, curr, next] = [null, head, null]
    while(curr){
         next = curr.next
         curr.next = prev
         prev = curr
         curr = next
   }
    return prev
};